osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacySwell.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Skinning;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Taiko.Objects.Drawables;
+using osu.Game.Rulesets.Taiko.Objects;
+using osu.Framework.Audio.Sample;
+using osu.Game.Audio;
+using osuTK;
+using osu.Game.Rulesets.Objects.Drawables;
+
+namespace osu.Game.Rulesets.Taiko.Skinning.Legacy
+{
+    public partial class LegacySwell : Container, ISkinnableSwell
+    {
+        private Container bodyContainer = null!;
+        private Sprite spinnerCircle = null!;
+        private Sprite shrinkingRing = null!;
+        private Sprite clearAnimation = null!;
+        private ISample? clearSample;
+        private LegacySpriteText remainingHitsCountdown = null!;
+
+        private bool samplePlayed;
+
+        public LegacySwell()
+        {
+            Anchor = Anchor.Centre;
+            Origin = Anchor.Centre;
+            RelativeSizeAxes = Axes.Both;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ISkinSource skin, SkinManager skinManager)
+        {
+            Child = new Container
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Position = new Vector2(200f, 100f),
+
+                Children = new Drawable[]
+                {
+                    bodyContainer = new Container
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Alpha = 0,
+
+                        Children = new Drawable[]
+                        {
+                            spinnerCircle = new Sprite
+                            {
+                                Texture = skin.GetTexture("spinner-circle"),
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Scale = new Vector2(0.8f),
+                            },
+                            shrinkingRing = new Sprite
+                            {
+                                Texture = skin.GetTexture("spinner-approachcircle") ?? skinManager.DefaultClassicSkin.GetTexture("spinner-approachcircle"),
```

Comment:
This fallback looks wrong. I think there's a high likelihood of this working wrong if a beatmap skin is missing `spinner-approachcircle` but the user skin isn't (doing this in such a way will cause the texture to fallback to default classic skin rather than use the user's skin).

Should be using `ISkinSource.FindProvider()`; many usages across the repository of this pattern.